### PR TITLE
[FEATURE] Cronjob to automatically create release PR

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Utilize a cron job to cut releases each week
  - Note that this is step 1 out of 2 or 3. This cron job simply creates the prep PR. The subsequent steps of tagging and creating release notes still needs to be automated.

ge_releaser tool: https://github.com/superconductive/ge_releaser
Sample prep PR: https://github.com/great-expectations/great_expectations/pull/5184

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: .github/workflows/release-prep.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        run: |
+          git clone https://github.com/superconductive/ge_releaser.git
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          pip install -e ge_releaser
+      - name: Create PR
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          ge_releaser prep --file=./.github/release_schedule.json
```
Comment (cdkini) [inline .github/workflows/release-prep.yml:24]: 1. Install the `ge_releaser` tool I use to manually cut releases each week
2. Read the `release_schedule.json` file and run `prep` on the version that matches the current day.

File: .github/release_schedule.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "2022-05-26": "0.15.7",
+    "2022-06-02": "0.15.8",
+    "2022-06-09": "0.15.9",
+    "2022-06-16": "0.15.10",
+    "2022-06-23": "0.15.11",
+    "2022-06-30": "0.15.12"
+}
```
Comment (cdkini) [inline .github/release_schedule.json:8]: This JSON configuration adheres to a particular schema as enforced by a test in `test_release.py`. It must be a set of key-value pairs with datetimes corresponding to semvers. The schedule must be non-empty, kept up-to-date, increase versions sequentially, etc. Any deviations will cause the test to fail and the build to exit early.

File: azure-pipelines-dependency-graph-testing.yml [hunk 3]
```diff
@@ -336,7 +336,8 @@ stages:
                 tests/datasource/test_sqlalchemy_datasource_workarounds.py \
                 tests/expectations/test_generate_diagnostic_checklist.py \
                 tests/test_deprecation.py \
-                tests/test_packaging.py
+                tests/test_packaging.py \
+                tests/test_release.py
```
Comment (cdkini) [inline azure-pipelines-dependency-graph-testing.yml:340]: Always run this test with each CI/CD cycle.

File: tests/test_release.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            # Each release occurs on a Thursday
+            assert date.weekday() == 3
+
+            curr_minor: int = release_version.minor
+            curr_patch: int = release_version.micro
+            prev_minor: int = prev_version.minor
+            prev_patch: int = prev_version.micro
+
+            # If incrementing a minor version number, the patch should be 0 (ex: 0.15.7 -> 0.16.0)
+            if curr_minor > prev_minor:
```
Comment (anthonyburdi) [inline tests/test_release.py:55]: I think Version also supports conditionals so you could also check prev_version < release_version.

**Review by anthonyburdi — APPROVED** [review]
LGTM! One small suggestion.

**Pull request merged**